src/example-apps/proxy/handlers/eventually_succeed.go

```diff
@@ -0,0 +1,47 @@
+package handlers
+
+import (
+	"log"
+	"net/http"
+	"os"
+	"strconv"
+	"sync"
+)
+
+type EventuallySucceedHandler struct {
+	callCount int
+	sync.RWMutex
+}
+
+const succeedAfterDefault = 5
+
+func (h *EventuallySucceedHandler) ServeHTTP(resp http.ResponseWriter, req *http.Request) {
+	logger := log.New(os.Stdout, "INFO: ", log.Ldate|log.Ltime)
+	cc := h.IncrementCallCount(logger)
+
+	if cc > getSuccessAfter() {
```

Comment:
Since the value is static for the runtime of the application (isn't it?) it would make sense to only call this during creation of the handler and store it in the handler.

Reply:
Good point. Fixed in this handler and in the similar eventually fail handler.

Reply:
The function to determine the amount of successful / failed responses is still called inside the handlers `ServeHTTP`, e.g.  it's evaluated every time the handler is called. I was more going for something like `NewExampleHandler` which does the lookup and returns an instance of the handler with the value already set.

Reply:
Ohhhh that makes more sense. Will fix when I get a chance.